src/scripts/main.js

```diff
@@ -1,3 +1,188 @@
 'use strict';
 
-// write code here
+let sortBy = null;
+const mainTable = document.querySelector('table');
+const newForm = document.createElement('form');
+
+newForm.classList.add('new-employee-form');
+
+const submitButton = document.createElement('button');
+
+submitButton.innerText = 'Save to table';
+
+const countryForSelect = [
+  'Tokyo',
+  'Singapore',
+  'London',
+  'New York',
+  'Edinburgh',
+  'San Francisco',
+];
+const allInputsForForm = [
+  createInput('text', 'Name'),
+  createInput('text', 'Position'),
+  createSelect(countryForSelect, 'Office'),
+  createInput('number', 'Age'),
+  createInput('number', 'Salary'),
+  submitButton,
+];
+
+newForm.append(...allInputsForForm);
+mainTable.after(newForm);
+
+function createInput(typeInput, nameInput) {
+  const newLabel = document.createElement('label');
+  const newInput = document.createElement('input');
+
+  newLabel.append(`${nameInput}:`);
+  newInput.name = nameInput.toLowerCase();
+  newInput.type = typeInput.toLowerCase();
+  newInput.dataset.qa = nameInput.toLowerCase();
+  newInput.required = true;
+  newLabel.append(newInput);
+
+  return newLabel;
+}
+
+function createSelect(array, nameForm) {
+  const newSelect = document.createElement('select');
+  const newLabel = document.createElement('label');
+
+  newLabel.append(`${nameForm}:`);
+
+  array.forEach((value) => {
+    const newOption = document.createElement('option');
+
+    newOption.value = value;
+    newOption.innerText = value;
+    newSelect.append(newOption);
+  });
+
+  newSelect.name = nameForm.toLowerCase();
+  newSelect.dataset.qa = nameForm.toLowerCase();
+  newLabel.append(newSelect);
+
+  return newLabel;
+}
+
+function sortForTable(sortTable, column) {
+  const tbody = sortTable.querySelector('tbody');
+
+  return Array.from(tbody.children).sort((row1, row2) => {
+    const value1 = row1.children[column].textContent;
+    const value2 = row2.children[column].textContent;
+
+    if (!/[0-9]/.test(value1)) {
+      return value1.localeCompare(value2);
+    } else if (/[0-9]/.test(value1)) {
+      return value1.replace(/\D/g, '') - value2.replace(/\D/g, '');
+    }
```

Comment:
The sorting logic in `sortForTable` may not handle numeric values correctly if they contain non-numeric characters. Consider ensuring that the values are properly parsed as numbers before sorting.